src/pybind/mgr/object_format.py

```diff
@@ -2,12 +2,71 @@
 # requested output formats such as JSON, YAML, etc.
 
 import enum
+import json
+
+from typing import Any, Optional
+
+import yaml
+
 
 class Format(enum.Enum):
-    plain = 'plain'
-    json = 'json'
-    json_pretty = 'json-pretty'
-    yaml = 'yaml'
-    xml_pretty = 'xml-pretty'
-    xml = 'xml'
+    plain = "plain"
+    json = "json"
+    json_pretty = "json-pretty"
+    yaml = "yaml"
+    xml_pretty = "xml-pretty"
+    xml = "xml"
+
+
+def _obj_to_json(obj: Any) -> Any:
+    tj = getattr(obj, "to_json", None)
+    if tj:
+        return tj()
+    return obj
+
+
+def _obj_to_yaml(obj: Any) -> Any:
+    if hasattr(obj, "yaml_representer"):
+        return obj
+    return _obj_to_json(obj)
```

Comment:
First, this change is largely meant to be a refactoring of existing code, so the point here is to reuse the existing approach to yaml formatting (my way to dodge blame for this ;-)). But as for the functionality, the idea is that *if* the obj has the yaml_representer attribute it "knows" how to turn itself into yaml and thus the original object is returned so that later code can make use of the yaml_representer. If the object is not "YAML aware" we fall back to getting a json friendly representation because *all* JSON can be represented as YAML, so its an easy fallback.